nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ScanAttribute.java

```diff
@@ -97,13 +107,23 @@
             .addValidator(StandardValidators.createRegexValidator(0, 1, false))
             .defaultValue(null)
             .build();
+    public static final PropertyDescriptor DICTIONARY_ENTRY_METADATA_DEMARCATOR = new PropertyDescriptor.Builder()
+            .name("Dictionary Entry Metadata Demarcator")
+            .description("A single character used to demarcate the dictionary entry string between dictionary value and metadata.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
If this is a single character, this validator should be a `StringLengthValidator` with a minimum and maximum of 1 character. See [here for an example](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteProcess.java#L129-L135).